solr/solrj/src/java/org/apache/solr/common/cloud/acl/ZkCredentialsInjector.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.common.cloud.acl;
+
+import java.util.List;
+
+public interface ZkCredentialsInjector {
+
+  List<ZkCredential> getZkCredentials();
+
+  class ZkCredential {
+
+    public enum Perms {
```

Comment:
Why do we need to use these instead of using ZooDefs.Perms directly?

Reply:
We need to be able to parse _perms_ from a human readable String, think for example if we are storing the creds in Json format in some Secret Manager and need to parse it back into a ZkCredential POJO object using data binding.
```
{
  "zkCredentials": [
                      {"username": "admin-user", "password": "ADMIN-PASSWORD", "perms": "all"},
                      {"username": "readonly-user", "password": "READONLY-PASSWORD", "perms": "read"}
                  ]
}
```